app/bundles/CoreBundle/Command/ModeratedCommand.php

```diff
@@ -180,7 +183,18 @@ private function checkPid(): bool
 
     private function checkFlock(): bool
     {
-        $store      = new FlockStore($this->runDirectory);
+        switch ($this->moderationMode) {
+            case self::MODE_REDIS:
+                $dsn   = $this->getContainer()->get('mautic.helper.core_parameters')->get('cache_adapter_redis')['dsn'] ?? null;
+                $redis = new \Redis();
+                $redis->connect(parse_url($dsn, PHP_URL_HOST));
```

Comment:
@kuzmany the DSN can contain a socket path, so the `parse_url` won't work in that case

Reply:
Good point 🥇 I have to check it again

Reply:
I rewrite it with support RedisAdapter , options and lock timeout